tests/model/test_ocrd_page.py

```diff
@@ -184,5 +184,39 @@ def test_empty_groups_to_regionrefindexed(self):
             children = og.get_AllIndexed()
             self.assertTrue(isinstance(children[1], RegionRefIndexedType))
 
+    def test_all_regions_without_reading_order(self):
+        """
+        https://github.com/OCR-D/core/pull/479
+        https://github.com/OCR-D/core/issues/240#issuecomment-493135797
+        """
+        with open('tests/model/TEMP1_Gutachten2-2.xml', 'r') as f:
+            pcgts = parseString(f.read().encode('utf8'), silence=True)
+            pg = pcgts.get_Page()
+            self.assertEqual(len(pg.get_AllRegions()), 45)
+            self.assertEqual(len(pg.get_AllRegions(depth=1)), 45)
+            self.assertEqual(len(pg.get_AllRegions(depth=2)), 65)
+            self.assertEqual(len(pg.get_AllRegions(depth=3)), 65)
```

Comment:
We should test infinite depth, too:
```suggestion
            self.assertEqual(len(pg.get_AllRegions(depth=3)), 65)
            self.assertEqual(len(pg.get_AllRegions(depth=0)), 65)
```

Reply:
But with `depth==0`, no recursion happens, return list would be empty.

Reply:
That's because you modified my original suggestion for how to _enter_ the loop from `depth + 1 if depth else 0` to just `depth + 1`.

But apparently this would not have worked, because there was also an infinity loop – see above.

Reply:
What was the purpose of `depth +1 if depth else 0`? The else clause would only be triggered if `depth == 0` or `depth == None`, right? So you want `depth == 0` to signify maximum depth, correct?

Reply:
> What was the purpose of `depth +1 if depth else 0`? 
> The else clause would only be triggered if `depth == 0` or `depth == None`, right? So you want `depth == 0` to signify maximum depth, correct?

Correct. The backend function discerns between always zero (termination only controlled by empty `more_regions`) and count-down (terminating at `level==1`). So the caller has to pass zero for infinite recursion but add an offset of 1 to ensure at least one level of recursion.

Reply:
That's erfect – thanks!